### ISSUE TITLE
Crashes on Android

### ISSUE THREAD

**Issue by ozz-rjq** [erickzanardo/ocarina#31]
When using `load()` method on Android, the app crashes with the following stacktrace:
```
Fatal Exception: java.lang.NoClassDefFoundError
Failed resolution of: Lcom/google/android/exoplayer2/source/ExtractorMediaSource;
xyz.erick.ocarina.AssetOcarinaPlayer.extractMediaSourceFromUri
```

```
Caused by java.lang.ClassNotFoundException
Didn't find class "com.google.android.exoplayer2.source.ExtractorMediaSource" on path: DexPathList[[zip file "/data/app/~~oMFEGBoML4tbSPBBEah5RQ==/app.gesso.android-vn5Sl0AXHKbWvevXMSpxQw==/base.apk"],nativeLibraryDirectories=[/data/app/~~oMFEGBoML4tbSPBBEah5RQ==/app.gesso.android-vn5Sl0AXHKbWvevXMSpxQw==/lib/x86, /data/app/~~oMFEGBoML4tbSPBBEah5RQ==/app.gesso.android-vn5Sl0AXHKbWvevXMSpxQw==/base.apk!/lib/x86, /system/lib, /system_ext/lib, /product/lib]]

dalvik.system.BaseDexClassLoader.findClass (BaseDexClassLoader.java:202)
java.lang.ClassLoader.loadClass (ClassLoader.java:312)

xyz.erick.ocarina.AssetOcarinaPlayer.extractMediaSourceFromUri (OcarinaPlugin.kt:131)
xyz.erick.ocarina.OcarinaPlayer.load (OcarinaPlugin.kt:89)
xyz.erick.ocarina.OcarinaPlugin.load (OcarinaPlugin.kt:209)
xyz.erick.ocarina.OcarinaPlugin.onMethodCall (OcarinaPlugin.kt:174)
io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage (MethodChannel.java:233)
io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart (DartMessenger.java:84)
io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage (FlutterJNI.java:822)
```

Seems like it may be caused by deprecation of [`ExtractorMediaSource`](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/source/ExtractorMediaSource.html)

**Comment by jmacpherson** [comment]
Have been debugging this same issue in one copy of my repo while another copy of my repo was not reproducing it. Managed to track it down to an issue resulting from usage of `flutter_html` which depends on `video_player`. In the repo that was not reproducing the issue, the dependency on `video_player` was resolving to `2.1.6`, whereas in the one that was reproducing the issue it was resolving to `2.1.10`.

Incidentally. it was in version `2.1.7` of `video_player` that they upgraded `ExoPlayer` `2.14.1`: https://pub.dev/packages/video_player/changelog

Specifying `video_player` `2.1.6` as a dependency override seems to have resolved the issue for the time being.

**Comment by erickzanardo** [comment]
Thanks for opening this issue. I have a lot on my plate right now and I will not have much time available to look into this in the short/medium future. Could use some help with a PR on this one.

**Comment by jmacpherson** [comment]
@erickzanardo It looks like you've already got the upgrade to ExoPlayer that is needed to resolve this issue, just that it's not yet released. I will test that it does resolve the issue my side and follow up.

**Comment by erickzanardo** [comment]
Oh that is right! I may have forgot to publish when I merged it.

Please let me know how your test went and I can publish a new version then.

**Comment by jmacpherson** [comment]
@erickzanardo Confirmed. The present `main` branch resolves the issue. I tested as follows:
1. On my copy of the repo that was not reproducing the issue, I overrode the `video_player` dep to `2.1.10`, this caused a reproduction of the issue
2. On the same copy, I then replaced the `ocarina` dep on `0.1.0` with one on this repo. The issue no longer occurred.

**Comment by ozz-rjq** [comment]
thanks!

**Comment by erickzanardo** [comment]
Thanks for confirming @jmacpherson 

I just published `0.1.1`.
